File: Game/Server/Include/ServerCore/Net/Packet/PacketObfuscator.cpp
#include "stdafx.h"

PacketObfuscator::PacketObfuscator() : 
	obfuscation(std::make_unique<XorObfuscation>()) {

}

PacketObfuscator::~PacketObfuscator() {

}

void PacketObfuscator::encodeHeader(Byte* src) {
	obfuscation->encode(src, 0, sizeof(int32_t));
}

void PacketObfuscator::encodeData(Byte* src, size_t len) {
	obfuscation->encode(src, sizeof(int32_t), len);
}

void PacketObfuscator::decodeHeader(Byte* src) {
	obfuscation->decode(src, 0, sizeof(int32_t));
}

void PacketObfuscator::decodeData(Byte* src, size_t len) {
	obfuscation->decode(src, sizeof(int32_t), len);
}

File: Game/Client/Assets/Scripts/Common/Net/PacketObfuscator.cs
using System;

namespace Heroes {
    public static class PacketObfuscator {
        private static Obfuscation obfuscation = new XorObfuscation();    

        public static void encodeHeader(ref Byte[] src) {
            obfuscation.encode(ref src, 0, sizeof(Int32));
        }

        public static void encodeData(ref Byte[] src, int len) {
            obfuscation.encode(ref src, sizeof(Int32), len);
        }

        public static void decodeHeader(ref Byte[] src) {
            obfuscation.decode(ref src, 0, sizeof(Int32));
        }

        public static void decodeData(ref Byte[] src, int len) {
            obfuscation.decode(ref src, sizeof(Int32), len);
        }
    }
}